v2/pkg/astvisitor/visitor.go

```diff
@@ -1365,19 +1410,24 @@ func (w *Walker) SetVisitorFilter(filter VisitorFilter) {
 
 // Walk initiates the walker to start walking the AST from the top root Node
 func (w *Walker) Walk(document, definition *ast.Document, report *operationreport.Report) {
-	if report == nil {
-		w.Report = &operationreport.Report{}
-	} else {
-		w.Report = report
-	}
-	w.Ancestors = w.Ancestors[:0]
-	w.Path = w.Path[:0]
-	w.TypeDefinitions = w.TypeDefinitions[:0]
-	w.document = document
-	w.definition = definition
-	w.Depth = 0
-	w.stop = false
-	w.walk()
+	ctx := context.Background()
+	labels := pprof.Labels("walker_id", w.walkerID)
+
+	pprof.Do(ctx, labels, func(ctx context.Context) {
```

Comment:
Are we okay with putting this into production?

Reply:
commented out